objects/src/lib.rs

```diff
@@ -55,117 +55,108 @@ pub mod utils {
+            // context.
+            while i < hex_bytes.len() {
+                // A hex-value is two characters per byte, and we need to reverse index to account
+                // for LE -> BE.
+                let ibyte = hex_bytes.len() - 1 - i;
+                let ifelt = felts.len() - 1 - (i / 2 / 8);
+
+                // This digit's nibble offset within the felt.
+                let inibble = i % (2 * 8);
+
+                // SAFETY: u8 cast to u64 is safe. We cannot use u64::from in const context so we
+                // are forced to cast.
+                let value = (parse_hex_digit(hex_bytes[ibyte]) as u64) << (inibble * 4);
+                felts[ifelt] += value;
+
+                i += 1;
             }
 
-            match hex_bytes.len() {
-                18 => {
-                    let v1 = u64::from_le_bytes(decode_u64_bytes(0, hex_bytes, 16));
-
-                    Digest::new([Felt::new(v1), Felt::new(0u64), Felt::new(0u64), Felt::new(0u64)])
-                },
-                34 => {
-                    let v1 = u64::from_le_bytes(decode_u64_bytes(0, hex_bytes, 16));
-                    let v2 = u64::from_le_bytes(decode_u64_bytes(16, hex_bytes, 32));
-
-                    Digest::new([Felt::new(v1), Felt::new(v2), Felt::new(0u64), Felt::new(0u64)])
-                },
-                50 => {
-                    let v1 = u64::from_le_bytes(decode_u64_bytes(0, hex_bytes, 16));
-                    let v2 = u64::from_le_bytes(decode_u64_bytes(16, hex_bytes, 32));
-                    let v3 = u64::from_le_bytes(decode_u64_bytes(32, hex_bytes, 48));
-
-                    Digest::new([Felt::new(v1), Felt::new(v2), Felt::new(v3), Felt::new(0u64)])
-                },
-                66 => {
-                    let v1 = u64::from_le_bytes(decode_u64_bytes(0, hex_bytes, 16));
-                    let v2 = u64::from_le_bytes(decode_u64_bytes(16, hex_bytes, 32));
-                    let v3 = u64::from_le_bytes(decode_u64_bytes(32, hex_bytes, 48));
-                    let v4 = u64::from_le_bytes(decode_u64_bytes(48, hex_bytes, 64));
-
-                    Digest::new([Felt::new(v1), Felt::new(v2), Felt::new(v3), Felt::new(v4)])
-                },
-                _ => panic!("Hex string has invalid length"),
-            }
+            $crate::Digest::new([
+                $crate::Felt::new(felts[0]),
+                $crate::Felt::new(felts[1]),
+                $crate::Felt::new(felts[2]),
+                $crate::Felt::new(felts[3]),
+            ])
         }};
     }
 
-    /// Test the correctness of the `digest!` macro for every supported hex string length.
-    #[test]
-    fn test_digest_macro() {
-        use crate::{Digest, Felt};
-
-        let digest_18_hex = "0x8B5563E13FE8135D";
-        let expected =
-            Digest::try_from("0x8B5563E13FE8135D000000000000000000000000000000000000000000000000")
-                .unwrap();
```

Comment:
I'm assuming this was actually incorrect behaviour? Should the result not have been

```rust
Digest::try_from("0x0000000000000000000000000000000000000000000000008B5563E13FE8135D")
```

Reply:
We had a conversation about endianness of the different hex strings ((here)[https://github.com/0xPolygonMiden/miden-vm/pull/1076#issuecomment-1741411558]), so the hex string which is used to create a Digest is assumed to be little-endian. Probably in that case it makes sense to assume that for the shorter hex strings too, since they are used to create a Digest as well. But that's a discussion matter, I'm not sure.

Reply:
Ah.. okay that was a confusing trip :) Thanks.